### DISCUSSION TITLE
ADD Ticketing ?

### DISCUSSION THREAD

**Discussion by Abygail007** [amidaware/tacticalrmm#1832] [Q&A]
Hello everyone !

I use Trmm and I love it (really!)

For my part, I find that it is only missing two major features:

The ability to manage mobile devices (android among others)
and ticketing management.

With that in addition to this rmm, I would stay connected day and night without ever sleeping again.

For ticketing I saw that this was integrated into trmm. Do you know how to install it?

https://github.com/conlan0/Trayicon

**Comment by conlan0** [comment]
Are you in the discord?

**Reply by Abygail007** [reply]
No sorry

**Reply by conlan0** [reply]
That's okay, I can help you add the tray icon to your setup if you'd like tho.

**Reply by Abygail007** [reply]
Yes with pleasure ! I configured the alerts.go file. Afterwards I don't know what to do.

**Reply by conlan0** [reply]
Okay awesome so if you did replace the apikey with a valid one from your rmm follow these steps,

1. Make sure you have golang downloaded on your computer you are building this with. https://go.dev/
2. Open a command prompt or terminal and navigate to where you downloaded and stored the Trayicon repository.
3. Run the command `go mod tidy` to install all the required packages.
4. Next run the command `go build -ldflags "-H=windowsgui"` to build the application.
5. Temporarily place the executable at a public facing url where it can be downloaded from.
6. Modify the script install.bat to work for your environment and add it to your rmm and than as an onboarding task to the agents or automation policy.
7. Done! Now when the task is created on the agent it will download and start launching the systray on boot and the agents will be able to submit "support tickets" that will get sent to the email recipients configured in global config and it will create alerts for each ticket that can be viewed in the alerts manager.

I will say that this is NOT an official implementation
If you decide to remove it from an agent or your TRMM it will need to be done manually or scripted
Official systray implementation is in the works but does not include support tickets for now

**Reply by Abygail007** [reply]
Thank you so much.

I'm stuck on:


Temporarily place the executable at a public facing url where it can be downloaded from. => the executable in .bat or an .exe? because I can't find the .exe


Modify the install.bat script to work for your environment and add it to your rmm and than as an onboarding task to the agents or automation policy.

I have to modify in the .bat: just after curl -o "tray.exe" -----
But I can't find the tray.exe after creating the application.

**Reply by conlan0** [reply]
So after running the build command there is no tray.exe in the trayicon folder?

Can you paste a picture of your terminal with those 2 command ouputs?

EDIT: i forgot to mention if you are building this on an os that is not windows you will need to run the command `set GOOS=windows` before running the command `go build -ldflags "-H=windowsgui"`

**Reply by conlan0** [reply]
But yes the exe is what needs to be placed at a url and yes thats the only part of the script that needs updated

**Reply by Abygail007** [reply]
![image](https://github.com/amidaware/tacticalrmm/assets/153526600/44581997-26bd-47af-a23c-3e86f1bbb6fe)

This is what I have but I can't find the .exe

**Reply by conlan0** [reply]
okay add me on discord so i can get back to you with faster replies and more personal support. my username is conlan0.


The reason you don't see the exe though is because it's not being built... the errors you're getting show it's trying to build for Linux and not windows and not being successful since it's a windows application.
You could also try this build command `env GOOS=windows GOARCH=amd64 go build -ldflags "-H=windowsgui"`

**Reply by conlan0** [reply]
Here you can see the steps i took to get it compiled.
![image](https://github.com/amidaware/tacticalrmm/assets/87742085/178d174c-3a83-4df6-b3da-f08c1f8798df)

**Reply by Abygail007** [reply]
Thanks, the tray.exe was compiled successfully. But on point 6 I modified the "install.bat", but how do I add it to the trmm?

I'm sorry I ask a lot of questions 🙃


Edit: sorry, it's all good! I was able to integrate it.

This creates an alert in my rmm, but no email.

This is a good basis for developing some real ticketing management in trmm.

**Reply by conlan0** [reply]
Awesome glad you got it working... Email alerts will go through but make sure you have all this info filled out in global settings in trmm.
![image](https://github.com/amidaware/tacticalrmm/assets/87742085/e2028038-e130-4113-a87e-082022129d61)

**Reply by Abygail007** [reply]
yes it is configured correctly. I already receive my usual alerts. Do I have a setting to make in trmm to link the alert to sending an email? because I created an alert for a Check but for the ticket there might be something to configure as well?

**Reply by conlan0** [reply]
Okay, nope there's no special place to enter the email creds for the systray but make sure you have the recipients in the global config and then it should just work. They will be the ones to receive the emails unlike how you configure the specific recipients for the alerts. If this is still the case where your not receiving the emails tho there might be a problem with how its handling sending them compared to my instance.

**Reply by Abygail007** [reply]
I receive the email notifications about the alert that I programmed and the global setting is correct. Weird weird 🤔🤔

![image](https://github.com/amidaware/tacticalrmm/assets/153526600/27249803-0414-4d5d-aa79-2dedff4f404d)
